src/Spaces/quadrature.jl

```diff
@@ -42,41 +42,135 @@ function quadrature_points end
 
 
 """
-    GLL{Nq}()
+    LegendreGaussLobatto
+    LGL
 
-Gauss-Legendre-Lobatto quadrature using `Nq` quadrature points.
+A Legendre-Gauss-Lobatto (LGL) quadrature rule.
+
+# Constructors
+
+    LegendreGaussLobatto(npoints=N)
+    LGL(npoints=N)
+
+Construct the quadrature rule with `N` quadrature points (`N ≥ 2`).
+
+    LegendreGaussLobatto(degree=p)
+    LGL(degree=p)
+
+Construct the quadrature rule with polynomial degree `p` (i.e. `p+1` quadrature
+points, with `p ≥ 1`).
+
+# Details
+
+A Legendre-Gauss-Lobatto rule has places quadrature points at the endpoints of
+the interval, and is able to exactly integrate polynomials of degree `2N-3`
+where `N` is the number of quadrature points.
+
+# External links
+
+- [Gauss-Lobatto rules on
+  Wikipedia](https://en.wikipedia.org/wiki/Gaussian_quadrature#Gauss%E2%80%93Lobatto_rules)
 """
-struct GLL{Nq} <: QuadratureStyle end
+struct LegendreGaussLobatto{Nq} <: QuadratureStyle end
+function LegendreGaussLobatto(;
+    npoints::Union{Int, Nothing} = nothing,
+    degree::Union{Int, Nothing} = nothing,
+)
+    if !isnothing(npoints)
+        LegendreGaussLobatto{npoints}()
+    elseif !isnothing(degree)
+        LegendreGaussLobatto{degree + 1}()
+    else
+        error("either npoints or degree option must be specified.")
+    end
+end
+const LGL = LegendreGaussLobatto
 
-polynomial_degree(::GLL{Nq}) where {Nq} = Nq - 1
-degrees_of_freedom(::GLL{Nq}) where {Nq} = Nq
+# TODO: deprecate
+const GLL = LegendreGaussLobatto
+polynomial_degree(::LegendreGaussLobatto{Nq}) where {Nq} = Nq - 1
+degrees_of_freedom(::LegendreGaussLobatto{Nq}) where {Nq} = Nq
 
-@generated function quadrature_points(::Type{FT}, ::GLL{Nq}) where {FT, Nq}
+@generated function quadrature_points(
+    ::Type{FT},
+    ::LegendreGaussLobatto{Nq},
+) where {FT, Nq}
     points, weights = GaussQuadrature.legendre(FT, Nq, GaussQuadrature.both)
```

Comment:
Since effectively this function only creates a tabulation of points and weights (as a quadrature rule should do), I stand by comment above, saying that defining this function with the `LegendreGaussLobatto(degree=p)` degree argument instead of the `qpoints` or `Nq`, however you want to call them, would be wrong in my opinion. A quadrature rule it's only an integration rule and does not need to know the polynomial order of the finite element basis